deploy/crds/app_v1alpha1_kogitoapp_crd.yaml

```diff
@@ -125,6 +125,15 @@ spec:
               description: The name of the runtime used, either quarkus or springboot,
                 defaults to quarkus
               type: string
+            service:
+              description: Service configuration
+              properties:
+                labels:
+                  additionalProperties:
+                    type: string
+                  description: Labels for the application service
+                  type: object
```

Comment:
I don't remember off the top of my head, but isn't there a better way to represent labels?

Reply:
@bmozaffa we’re using map[string]string for representing the Labels object.

Reply:
Right, I mean let's find the right way to represent it in CRD too, so it's validated and now free form.

Reply:
@bmozaffa please find this example of [string to string mapping](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#model-with-mapdictionary-properties) for OpenAPI which is the same as what we do. This yaml is auto-generated from the [kogitoapp_types](https://github.com/kiegroup/kogito-cloud-operator/blob/d7b0a6214eb55359e66004c93227bbc33a1ef87a/pkg/apis/app/v1alpha1/kogitoapp_types.go#L80) definition, and the kogitoapp_types definition is based on the type definition from [apimachinery](https://github.com/kubernetes/apimachinery/blob/533d101be9a6450773bb2829bef282b6b7c4ff6d/pkg/apis/meta/v1/types.go#L234) code. Thanks.

Reply:
Thanks, Xieshen, for looking into it. The CRD generation is not always the best, but if we don't have any example of trying to describe and enforce labels anywhere, whether possible or not, probably not worth trying to do it.

Reply:
@bmozaffa The [validation](https://github.com/kiegroup/kogito-cloud-operator/blob/d7b0a6214eb55359e66004c93227bbc33a1ef87a/deploy/crds/app_v1alpha1_kogitoapp_crd.yaml#L14) is via [OpenAPI V3 Schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation). In the OpenAPI V3 Schema Object [example](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#model-with-mapdictionary-properties), string to string mapping is like what we do here.